### PR TITLE
Add subscriptions handling

### PR DESCRIPTION
**Pull request by radufa**
**Description**
The Knative access library was extended to contain the handling of Knative subscriptions.
 
**Related issue(s)**

#2298

### CODE REVIEW

File: components/event-bus/internal/knative/util/eventing.go [hunk 1]
```diff
@@ -85,6 +86,9 @@ func (k *KnativeLib) GetChannel(name string, namespace string) (*evapisv1alpha1.
 		log.Printf("ERROR: GetChannel(): geting channel: %v", err)
 		return nil, err
 	} else {
```
Comment (suleymanakbas91) [inline components/event-bus/internal/knative/util/eventing.go]: If already has a `return`, so you don't need to use `else` here.

File: components/event-bus/internal/knative/util/eventing.go [hunk 2]
```diff
@@ -115,21 +119,59 @@ func ( k*KnativeLib) CreateChannel(provisioner string, name string, namespace st
...
 		if sub, err = k.evClient.Subscriptions(namespace).Update(sub); err != nil {
-			log.Printf("ERROR: CreateSubscription(): update subscription: %v", err)
+			log.Printf("ERROR: CreateSubscription(): updating subscription: %v", err)
 			return err
 		}
 	}
 	return nil
 }
 
+// DeleteSubscription deletes a Kantive/Eventing subscription
```
Comment (suleymanakbas91) [inline components/event-bus/internal/knative/util/eventing.go]: Typo: `Kantive` should be `Knative`

File: components/event-bus/internal/knative/util/eventing.go [hunk 3]
```diff
@@ -115,21 +119,59 @@ func ( k*KnativeLib) CreateChannel(provisioner string, name string, namespace st
...
+		log.Printf("ERROR: DeleteSubscription(): deleting subscription: %v", err)
+		return err
+	}
+	return nil
+}
+
+// GetSubscription gets a Knative/Eventing subscription
+func (k *KnativeLib) GetSubscription(name string, namespace string) (*evapisv1alpha1.Subscription, error) {
+	if sub, err := k.evClient.Subscriptions(namespace).Get(name, metav1.GetOptions{}); err != nil {
+		log.Printf("ERROR: GetSubscription(): geting subscription: %v", err)
```
Comment (suleymanakbas91) [inline components/event-bus/internal/knative/util/eventing.go]: `geting` should be `getting`

File: components/event-bus/internal/knative/util/eventing.go [hunk 4]
```diff
@@ -115,21 +119,59 @@ func ( k*KnativeLib) CreateChannel(provisioner string, name string, namespace st
...
+		return nil, err
+	} else {
```
Comment (suleymanakbas91) [inline components/event-bus/internal/knative/util/eventing.go]: `else` is redundant.

File: components/event-bus/internal/knative/util/eventing.go [hunk 5]
```diff
@@ -115,21 +119,59 @@ func ( k*KnativeLib) CreateChannel(provisioner string, name string, namespace st
...
+		return sub, nil
+	}
+}
+
+// UpdateSubscription updates an existing subscription
+func (k *KnativeLib) UpdateSubscription(sub *evapisv1alpha1.Subscription) (*evapisv1alpha1.Subscription, error) {
+	if usub, err := k.evClient.Subscriptions(sub.Namespace).Update(sub); err != nil {
+		log.Printf("ERROR: UpdateSubscription(): updating subscription: %v", err)
+		return nil, err
+	} else {
```
Comment (suleymanakbas91) [inline components/event-bus/internal/knative/util/eventing.go]: `else` is redundant.

File: components/event-bus/internal/knative/util/eventing.go (hunk 1 above)
Comment (radufa) [reply components/event-bus/internal/knative/util/eventing.go]: the "channel" is bound to the "if" context

File: components/event-bus/internal/knative/util/eventing.go (hunk 2 above)
Comment (radufa) [reply components/event-bus/internal/knative/util/eventing.go]: Done

File: components/event-bus/internal/knative/util/eventing.go (hunk 3 above)
Comment (radufa) [reply components/event-bus/internal/knative/util/eventing.go]: Done

**Pull request merged**